Feat/#23 post like

hyewonbae opened the pull request:
# PR Description
> 1. domain 패키지 분리작업
> 2. 게시물 좋아요 기능 구현 (like)
분산락과 스프링 스케줄러 사용
redis 디비 --> mysql 디비 동기화 작업
동기화처리 임시 주석(LikeSyncScheduler.java / line:28 )

## Related Issue

close # 23

## Help
> 아쉬운 점
1. 레디스 캐시 이용해보는 것
2. 동기화 작업 후 레디스 데이터 삭제하는 것 
위 두 사안에 대해서 반영하려면 현재 작업했던 동기화 로직 전체 수정이 필요한데,
현재 소스 수정하기엔 시간적 여유도 없고 발생하는 리스크가 너무 크다고 생각되어 수정하진 않았습니다. 다만, 어떤 문제가 발생할 수 있으며 고려해야하는 점이 무엇인지는 인지한 상태여서 추후 다른 방안에 대해서 공부해보도록 하겠습니다.

hyewonbae committed:
feat : create post like

- 게시물 좋아요 기능구현
+ refact : move entity classes within domain package
- 도메인 패키지 안의 엔티티 클래스 이동 리팩토링

hyewonbae committed:
feat : add sync logic

동기화 로직 추가 및 리팩토링

weejinyoung commented on src/main/java/mini/backend/like/Like.java:
```diff
@@ -0,0 +1,33 @@
+        @UniqueConstraint(columnNames = {"user_id", "post_id"})
+})
+public class Like extends BaseTimeEntity {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long like_id;
```
문서상으로 정해진 것은 아니지만 우리 프로젝트의 네이밍 컨벤션은 보편적인 자바 네이밍 컨벤션을 많이 따르는 것으로 이해하고 있어요
지금까지 변수 네이밍은 카멜 케이스를 따르고 있는데 요 변수는 스네이크 케이스를 따르고 있네요 혹시 요 변수만 따로 스네이크 케이스로 하신 이유가 있을까요? 혹시 JPA Entity 규칙 때문에 이렇게 하신 거라면 column 어노테이션에 옵션을 줘서 컬럼 이름을 정해주는 게 어떨까 싶어요

hyewonbae replied:
엔티티 생성하면서 습관적으로 스네이크 케이스로 썼던 것 같습니다. 요 부분 수정해서 반영하겠습니다~

weejinyoung commented on src/main/java/mini/backend/like/LikeService.java:
```diff
@@ -0,0 +1,57 @@
+                // 락 해제
+                redisTemplate.delete(lockKey);
+            }
+        } else {
+            // 락 대기 상태에서 재시도 로직 추가 가능
+            return addLike(postId, Id); // 재귀 호출 (주의: 무한 루프 방지)
+        }
```
5초의 lease time 동안 여러 서버에서 여러 스레드가 while 문을 돌며 락 재점유 시도 시 발생할 수 있는 스택오버플로우 문제 등등...
지난 오전에 같이 이야기 나누었던 부분들 꼭 고민해보셨으면 해요!
너무나 좋은 소재들이라..

hyewonbae replied:
넵~ 공부할 때 추가로 해보겠습니다.

hyewonbae committed:
fix : Solving FK Constraint Issues

1. post (부모)테이블에서 레코드 삭제 시 참조하는 comment (자식)테이블 레코드 선삭제

2. Like entity naming
snake case -> camel case

PeindreLaColline commented:
수고하셨읍니다,,,,

prin81208 commented:
확인했습니다. 수고하셨습니다 !
